tests/js/common/replication_api/shell-api-replication.js

```diff
@@ -0,0 +1,2524 @@
+        assertMatch(/^[a-zA-Z0-9_\-]+$/, doc['data']['_rev']);
+        assertTrue(doc['data'].hasOwnProperty('test'));
+
+        body = body.slice(position + 1, body.length);
+        i = i + 1;
+      }
+      assertEqual(i, 100);
+    },
+
+    test_fetches_incremental_parts_of_a_collection_dump_1: function() {
+      let cid = db._create("UnitTestsReplication");
+
+      for (let i = 0; i < 10; i++) {
+        let body = { "_key" : `test${i}`, "test" : `${i}` };
+        let doc = arango.POST_RAW("/_api/document?collection=UnitTestsReplication", body);
+        assertEqual(doc.code, 202);
+      }
+
+      let doc = arango.PUT_RAW("/_admin/wal/flush?waitForSync=true&waitForCollector=true", "");
+      assertEqual(doc.code, 200);
+
+      let fromTick = 0;
+
+      arango.DELETE_RAW(api + `/batch/${batchId}`, "");
+      let doc0 = arango.POST_RAW(api + "/batch", {});
+      batchId = doc0.parsedBody["id"];
+      assertMatch(/^[0-9]+$/, batchId);
+
+      for (let i = 0; i < 10; i++) {
+        let cmd = api + `/dump?collection=UnitTestsReplication&from=${fromTick}&chunkSize=1&batchId=${batchId}`;
+        doc = arango.GET_RAW(cmd);
+        assertEqual(doc.code, 200);
+
+        if (i === 9) {
+          assertEqual(doc.headers["x-arango-replication-checkmore"], "false");
+        } else {
+          assertEqual(doc.headers["x-arango-replication-checkmore"], "true");
+        }
+
+        assertMatch(/^[0-9]+$/, doc.headers["x-arango-replication-lastincluded"]);
+        assertNotEqual(doc.headers["x-arango-replication-lastincluded"], "0");
+        assertTrue(Number(doc.headers["x-arango-replication-lastincluded"]) >= Number(fromTick));
+        assertEqual(doc.headers["content-type"], "application/x-arango-dump");
+
+        fromTick = doc.headers["x-arango-replication-lastincluded"];
+
+        let document = JSON.parse(doc.body.toString());
+        assertEqual(document['type'], 2300);
+        assertMatch(/^test[0-9]+$/, document['data']['_key']);
+        assertMatch(/^[a-zA-Z0-9_\-]+$/, document['data']['_rev'], document);
+        assertTrue(document['data'].hasOwnProperty('test'));
+      }
+    }
+  };
+}
+
+function dealing_with_general_functionSuite () {
+  let api = "/_db/UnitTestDB/_api/replication";
+  return {
+    setUp: function() {
+      let res = db._createDatabase("UnitTestDB");;
+      assertTrue(res);
+    },
+
+    tearDown: function() {
+      let res = db._dropDatabase("UnitTestDB");;
+      assertTrue(res);
+    },
+    test_fetches_the_server_id_2: function() {
+      // fetch id;
+      let cmd = api + "/server-id";
+      let doc = arango.GET_RAW(cmd);
+
+      assertEqual(doc.code, 200);
+      assertMatch(/^[0-9]+$/, doc.parsedBody['serverId']);
+    }
+  };
+}
+////////////////////////////////////////////////////////////////////////////////;
+// applier;
+////////////////////////////////////////////////////////////////////////////////;
+function dealing_with_the_applieSuite () {
```

Comment:
It looks like this suite is already mostly present above (https://github.com/arangodb/arangodb/pull/15935/files#diff-d30ab47493830791c04c990e1780dee036b030ecbe3fa694fc8903ccec338089R57 and following).
The difference between the 2 suites seems to be that one of them is using the `_system` database, and the other one is using a different database.
But it looks like the code is mostly redundant.

Reply:
we decided not to approach to de-duplicate the test.